src/OgAccess.php

```diff
@@ -0,0 +1,278 @@
+   *
+   * @var string
+   */
+  const NEUTRAL = NULL;
+
+
+
+  /**
+   * Determines whether a user has a given privilege.
+   *
+   * All permission checks in OG should go through this function. This
+   * way, we guarantee consistent behavior, and ensure that the superuser
+   * and group administrators can perform all actions.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $group_entity
+   *   The group entity.
+   * @param string $operation
+   *   The entity operation being checked for.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   (optional) The user to check. Defaults to the current user.
+   * @param $skip_alter
+   *   (optional) If TRUE then user access will not be sent to other modules
+   *   using drupal_alter(). This can be used by modules implementing
+   *   hook_og_user_access_alter() that still want to use og_user_access(), but
+   *   without causing a recursion. Defaults to FALSE.
+   * @param $ignore_admin
+   *   (optional) When TRUE the specific permission is checked, ignoring the
+   *   "administer group" permission if the user has it. When FALSE, a user
+   *   with "administer group" will be granted all permissions.
+   *   Defaults to FALSE.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   TRUE or FALSE if the current user has the requested permission.
+   *   NULL, if the given group isn't a valid group.
+   */
+  public static function userAccess(EntityInterface $group_entity, $operation, AccountInterface $user = NULL, $skip_alter = FALSE, $ignore_admin = FALSE) {
+    $group_type_id = $group_entity->getEntityTypeId();
+    $bundle = $group_entity->bundle();
+
+    if (!Og::isGroup($group_type_id, $bundle)) {
+      // Not a group.
+      return AccessResult::neutral()->addCacheableDependency($group_entity);
+    }
+
+    $user = $user ?: \Drupal::currentUser()->getAccount();
+    $user_id = $user->id();
+
+    // User ID 1 has all privileges.
+    if ($user_id == 1) {
+      return AccessResult::allowed();
+    }
+
+    // Administer group permission.
+    if (!$ignore_admin) {
+      $user_access = AccessResult::allowedIfHasPermission($user, static::ADMINISTER_GROUP_PERMISSION);
+      if ($user_access->isAllowed()) {
+        return $user_access;
+      }
+    }
+
+    // Group manager has all privileges (if variable is TRUE) and they are
+    // authenticated.
+    $config = \Drupal::config('og.settings');
+    if ($config->get('group_manager_full_access')) {
+      if (!empty($user_id) && $group_entity instanceof EntityOwnerInterface && $group_entity->getOwnerId() == $user_id) {
+        // @TODO do we need to add the user here?
+        return AccessResult::allowed()
+          ->addCacheableDependency($config);
+      }
+    }
+
+    $pre_alter_cache = [];//static::getPermissionsCache($group_entity, $user, TRUE);
+    $post_alter_cache = [];//static::getPermissionsCache($group_entity, $user, FALSE);
+
```

Comment:
Indeed, right now they are empty - they will be filled once we port the OgRoles and permissions (the `@todo` few lines below)